### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by SridharJagannathan**
Updated for the issue #110. The relevant commits have the specific comments. To summarise this fixes the issue of environmental variables not working in linux machines (when using the check command in R). Also I fixed the test cases in tests/testthat/test-connections.R to point to a public server and added test cases for the set and get env vars. However I have not added any tests cases for the unset category as I'm not sure the need for that function itself.

### CODE REVIEW

File: R/catmaid_connection.R [hunk 1]
```diff
@@ -441,12 +445,12 @@ catmaid_connection_fingerprint<-function(conn) {
 #' @seealso \code{\link{catmaid_login}}
 #' @export
 catmaid_connection_setenv<-function(conn=NULL, ...) {
-  conn=catmaid_login(conn, ...)
+  if (is.null(conn)){conn=catmaid_login(conn, ...)} #only do the connection if I supply a null..
```
Comment (jefferis) [inline R/catmaid_connection.R]: `catmaid_login` looks after checking the value of conn already – so just call it as before i.e. revert this change.
```suggestion
 conn=catmaid_login(conn, ...)
```

File: R/catmaid_connection.R [hunk 2]
```diff
@@ -534,3 +538,23 @@ set_catmaid_version <- function(version, conn=NULL, ...) {
...
+  matchvalues <- grep(pattern = tempstr, x = names(Sys.getenv()), value = TRUE)
+  
+  if (length(grep(pattern = paste0("^", "catmaid", "\\.") , x= matchvalues)) == 4) {
+    return(paste0("catmaid."))
+  } else if (length(grep(pattern = paste0("^", "catmaid", "_") , x= matchvalues)) == 4) {
+    return(paste0("catmaid_"))
+  }else if (length(grep(pattern = paste0("^", "catmaid", "\\.") , x= matchvalues)) == 0 && 
+            length(grep(pattern = paste0("^", "catmaid", "_") , x= matchvalues)) == 0) {
+    stop(paste("\ncatmaid error: No usable environmental variables found"))
+  }
```
Comment (jefferis) [inline R/catmaid_connection.R]: Don't error out here – if you look at travis you'll see that this code fails in the absence of appopriately defined environment variables.

File: R/catmaid_connection.R [hunk 3]
```diff
@@ -129,6 +130,9 @@
 #'   config=conn$config)
 #' }
 #' @export
+#' 
+#' @examples
+#' conn=catmaid_login()
```
Comment (jefferis) [inline R/catmaid_connection.R]: Please don't add an unprotected example like this. It is not guaranteed that an end user will be able to call catmaid_login without any arguments since the may not have setup .Renviron or .Rprofile appropriately.

File: R/catmaid_connection.R (hunk 2 above)
Comment (jefferis) [reply R/catmaid_connection.R]: Also 
```
stop("catmaid error: No usable environmental variables found")
```
should be ok. The paste isn't doing anything. Also why the leading `\n`?

File: tests/testthat/test-connections.R [hunk 4]
```diff
@@ -1,5 +1,27 @@
 context("catmaid login and get/post")
 
+test_that("set and get the environmental variables responsible for connections", {
+  #check the named object has all necessary fields..
+  expect_named(catmaid_connection_getenv(), c("server", "authname", "authpassword","token"))
```
Comment (jefferis) [inline tests/testthat/test-connections.R]: Will this work if no environment variables are set?

File: tests/testthat/test-connections.R [hunk 5]
```diff
@@ -27,21 +49,32 @@ test_that("can make a connection from list elements", {
 })
 
 test_that("can login", {
+  #Just restart the connection here to a public server..
+  conn <- catmaid_login(server='http://hildebrand16.neurodata.io/catmaid/', Cache = FALSE)
+
```
Comment (jefferis) [inline tests/testthat/test-connections.R]: I would roll back all of these changes for now.

Comment (jefferis) [reply tests/testthat/test-connections.R]: i.e. the rest of this file.

File: R/catmaid_connection.R (hunk 1 above)
Comment (SridharJagannathan) [reply R/catmaid_connection.R]: I have updated this in the latest commit in line 446.

File: R/catmaid_connection.R (hunk 2 above)
Comment (SridharJagannathan) [reply R/catmaid_connection.R]: I have updated this as a simplemessage in line 552 in the latest commit and I have also added testcases for the same in the tests.

File: R/catmaid_connection.R (hunk 3 above)
Comment (SridharJagannathan) [reply R/catmaid_connection.R]: I have removed this in the latest commit, that being said the reason I added this is that the bug is reproducible only if you have an example running. So we need to have such an example in the future.

File: tests/testthat/test-connections.R (hunk 5 above)
Comment (SridharJagannathan) [reply tests/testthat/test-connections.R]: I have rolled back these changes.

File: tests/testthat/test-connections.R (hunk 4 above)
Comment (SridharJagannathan) [reply tests/testthat/test-connections.R]: In the latest commit, I have modified the test cases, such that the case will only run if all the env vars are set otherwise it skips the case with a message. I have also added a new object that is at the top where we can easily configure the public server url in the future.
